NPEs on workspace start-up

### Description
Several NPEs might happen on workspace startup when an JsonRpcError instance is created from an Exception with no message provided (for example, TimeoutException).

### Reproduction Steps
- Create Java Workspace
- I have created a Plain Java project and open in editor its Main.java class
- Stop and restart workspace.
- Watch dev-machine log during the workspace start up

![lsnpe](https://user-images.githubusercontent.com/620781/41534936-b049fefe-7300-11e8-9e2f-37fadb8f9a05.gif)

**OS and version:**    

**Diagnostics:** 
```
2018-06-18 11:47:12,903[equestProcessor]  [ERROR] [oggingUncaughtExceptionHandler 31]   - Runtime exception caught in thread ServerSideRequestProcessor. Message: null
java.lang.NullPointerException: null
	at org.eclipse.che.api.core.jsonrpc.impl.GsonJsonRpcMarshaller.marshall(GsonJsonRpcMarshaller.java:46)
	at com.google.gson.JsonPrimitive.setValue(JsonPrimitive.java:105)
	at com.google.gson.JsonPrimitive.isPrimitiveOrString(JsonPrimitive.java:282)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at org.eclipse.che.api.core.jsonrpc.commons.JsonRpcErrorTransmitter.transmit(JsonRpcErrorTransmitter.java:45)
	at com.google.gson.JsonPrimitive.<init>(JsonPrimitive.java:65)
	at org.eclipse.che.api.core.jsonrpc.commons.JsonRpcMessageReceiver.processRequest(JsonRpcMessageReceiver.java:102)
	at org.eclipse.che.api.core.jsonrpc.impl.GsonJsonRpcMarshaller.getError(GsonJsonRpcMarshaller.java:117)
	at java.lang.Thread.run(Thread.java:748)
	at org.eclipse.che.api.core.jsonrpc.commons.JsonRpcMessageReceiver.lambda$receive$0(JsonRpcMessageReceiver.java:72)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.NullPointerException: null
	at com.google.gson.JsonPrimitive.setValue(JsonPrimitive.java:105)
	at com.google.gson.JsonPrimitive.<init>(JsonPrimitive.java:65)
2018-06-18 11:47:16,682[equestProcessor]  [ERROR] [oggingUncaughtExceptionHandler 31]   - Runtime exception caught in thread ServerSideRequestProcessor. Message: null
	at org.eclipse.che.api.core.jsonrpc.commons.JsonRpcMessageReceiver.processRequest(JsonRpcMessageReceiver.java:102)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at com.google.gson.JsonPrimitive.isPrimitiveOrString(JsonPrimitive.java:282)
	at org.eclipse.che.api.core.jsonrpc.commons.JsonRpcErrorTransmitter.transmit(JsonRpcErrorTransmitter.java:45)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.eclipse.che.api.core.jsonrpc.commons.JsonRpcMessageReceiver.lambda$receive$0(JsonRpcMessageReceiver.java:72)
	at org.eclipse.che.api.core.jsonrpc.impl.GsonJsonRpcMarshaller.marshall(GsonJsonRpcMarshaller.java:46)
	at org.eclipse.che.api.core.jsonrpc.impl.GsonJsonRpcMarshaller.getError(GsonJsonRpcMarshaller.java:117)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcError.java
+++ b/core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcError.java
@@ -10,12 +10,14 @@
  */
 package org.eclipse.che.api.core.jsonrpc.commons;
 
+import javax.validation.constraints.NotNull;
+
 /** Represents JSON RPC error object */
 public class JsonRpcError {
   private final int code;
   private final String message;
 
-  public JsonRpcError(int code, String message) {
+  public JsonRpcError(int code, @NotNull String message) {
     this.code = code;
     this.message = message;
   }

--- a/core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcErrorTransmitter.java
+++ b/core/che-core-api-core/src/main/java/org/eclipse/che/api/core/jsonrpc/commons/JsonRpcErrorTransmitter.java
@@ -40,7 +40,8 @@ public class JsonRpcErrorTransmitter {
 
     LOGGER.debug("Transmitting a JSON RPC error: " + e.getMessage());
 
-    JsonRpcError error = new JsonRpcError(e.getCode(), e.getMessage());
+    JsonRpcError error =
+        new JsonRpcError(e.getCode(), e.getMessage() == null ? "Unexpected error" : e.getMessage());
     JsonRpcResponse response = new JsonRpcResponse(e.getId(), null, error);
     String message = marshaller.marshall(response);
     transmitter.transmit(endpointId, message);

